src/SDL.c

```diff
@@ -107,6 +107,52 @@ SDL_NORETURN void SDL_ExitProcess(int exitcode)
 #endif
 }
 
+/* App metadata */
+static SDL_PropertiesID SDL_app_metadata_properties = 0;
+
+int SDL_SetAppMetadata(const char *appname, const char *appversion, const char *appidentifier)
+{
+    const SDL_PropertiesID props = SDL_CreateProperties();
+
+    if (appname && *appname) {
+        SDL_SetStringProperty(props, SDL_PROP_APP_METADATA_NAME_STRING, appname);
+    }
+    if (appversion && *appversion) {
+        SDL_SetStringProperty(props, SDL_PROP_APP_METADATA_VERSION_STRING, appversion);
+    }
+    if (appidentifier && *appidentifier) {
+        SDL_SetStringProperty(props, SDL_PROP_APP_METADATA_IDENTIFIER_STRING, appidentifier);
+    }
+
+    const int retval = SDL_SetAppMetadataWithProperties(props);
+    SDL_DestroyProperties(props);
+    return retval;
+}
+
+int SDL_SetAppMetadataWithProperties(SDL_PropertiesID props)
+{
+    SDL_PropertiesID new_props = 0;
+    if (props) {
+        new_props = SDL_CreateProperties();
+        if (!new_props) {
+            return -1;
+        } else if (SDL_CopyProperties(props, new_props) == -1) {
```

Comment:
```suggestion
        } else if (SDL_CopyProperties(props, new_props) < 0) {
```
For consistency with the rest of the codebase

Reply:
Yes, but I feel like I've dumped this all over the codebase, I should grep for that.

Reply:
Fixed in 164928aff9b82ae3a43940b0dbe41307439fbaba.